source/code-snippets/transactions/txn-convenient.js

```diff
@@ -0,0 +1,83 @@
+import { MongoError, MongoClient, ObjectId } from "mongodb";
+
+const uri =
+  "<connection string uri>";
+const client = new MongoClient(uri);
+await client.connect();
+
+await client
+  .db("testdb")
+  .collection("inventory")
+  .insertMany([
+    // start-inventory
+    { item: "sunblock, 8 oz", qty: 85, price: 6.0 },
+    { item: "beach chair", qty: 30, price: 25.0 }
+    // end-inventory
+  ]);
+
+const order1 = [
+// start-order-successful
+  { item: "sunblock, 8 oz", qty: 3 },
+  { item: "beach chair", qty: 1 }
+// end-order-successful
+];
+
+const order2 = [
+// start-order-fail
+  { item: "volleyball", qty: 1 }
+// end-order-fail
+];
+
+let order = order1;
+
+// start-transaction
+const txnResult = await client.withSession(async (session) =>
+  session
+    .withTransaction(async (session) => {
+      const invColl = client.db("testdb").collection("inventory");
+      const recColl = client.db("testdb").collection("records");
+
+      let total = 0;
+      for (const item of order) {
+        // Abort the transaction if the item
+        // does not exist or has insufficient inventory
+        const checkInventory = await invColl.findOne(
+          {
+            item: item.item,
+            qty: { $gte: item.qty },
+          },
+          { session }
+        );
+        if (checkInventory === null) {
+          await session.abortTransaction();
+          return "Item not found or insufficient quantity.";
+        }
+
+        // If inventory is sufficient, update inventory
+        // to account for the purchase.
+        await invColl.updateOne(
+          { item: item.item },
+          { $inc: { qty: -item.qty } },
+          { session }
+        );
```

Comment:
Yes, the operations are separated because the transaction should abort if no matching documents are found (aka insufficient quantity). I guess we still could combine this in one compound operation, and just abort the transaction based on the return type of `findOneAndUpdate()`, but I think separating them makes things more clear for the user to understand

Reply:
I think splitting the operations makes it the example unnecessarily long and might cause confusion as to why the find is separated from update. I think it would make more sense to use the compound operation. I think the insert into the inventory collection sufficiently demonstrates a second operation in the transaction since it can't be combined in a single compound operation.